### PR TITLE
It would appear that the first Chunk is always discarded, this fixes …

### PR DESCRIPTION
**Pull request by viktorklang**
…that

### CODE REVIEW

**Comment by sleipnir** [comment]
It seems worthwhile to open an issue to the AKKA repository for research right?

**Comment by viktorklang** [comment]
@sleipnir Not really, this is our code (Cloudstate).

**Comment by sleipnir** [comment]
Okay @viktorklang I proposed this because I already had problems using FileIO where it dropped some messages, but I have no evidence of that at the moment, so I thought it could be something more profound.

**Comment by viktorklang** [comment]
@sleipnir Check if it fares better with this PR :)

**Comment by ralphlaude** [comment]
@viktorklang, cool. How did you find it out?

**Comment by viktorklang** [comment]
@ralphlaude Pure luck, while reading code.

**Comment by pvlugter** [comment]
So this is not tested in any way?

**Comment by viktorklang** [comment]
@pvlugter Let me check, perhaps it is not checking the right thing.

**Comment by raboof** [comment]
I'll port this one since it conflicts with #312

File: proxy/core/src/main/scala/io/cloudstate/proxy/HttpApi.scala [hunk 1]
```diff
@@ -527,22 +527,26 @@ object HttpApi {
...
 
-              val data =
-                if (isHttpBodyResponse) extractDataFromHttpBody(entityMessage)
-                else ByteString(jsonPrinter.print(entityMessage))
+              val first = {
+                val data =
+                  if (isHttpBodyResponse) extractDataFromHttpBody(entityMessage)
+                  else ByteString(jsonPrinter.print(entityMessage))
+
+                Source.single(HttpEntity.Chunk(data))
```
Comment (raboof) [inline proxy/core/src/main/scala/io/cloudstate/proxy/HttpApi.scala:535]: no newline here

File: proxy/core/src/main/scala/io/cloudstate/proxy/HttpApi.scala [hunk 2]
```diff
@@ -527,22 +527,26 @@ object HttpApi {
...
               HttpResponse(
                 entity = HttpEntity.Chunked(
                   contentType,
-                  rest map { protobuf: ProtobufAny =>
+                  first.concat(rest map { protobuf: ProtobufAny =>
                     val entityMessage: MessageOrBuilder = parseResponseBody(Serve.ReplySerializer.serialize(protobuf))
                     val data =
                       if (isHttpBodyResponse) extractDataFromHttpBody(entityMessage)
                       else ByteString(jsonPrinter.print(entityMessage))
                     HttpEntity.Chunk(data ++ NEWLINE_BYTES)
```
Comment (raboof) [inline proxy/core/src/main/scala/io/cloudstate/proxy/HttpApi.scala:548]: but a newline here. is that OK?

File: proxy/core/src/main/scala/io/cloudstate/proxy/HttpApi.scala (hunk 1 above)
Comment (viktorklang) [reply proxy/core/src/main/scala/io/cloudstate/proxy/HttpApi.scala:535]: Good catch—addressing

**Pull request merged**